library/game_state.py

```diff
@@ -155,9 +198,25 @@ def confirm_good_move(self, board: Board) -> bool:
 
         return self.good_move
 
+    def confirm_good_subgrid(self, board: Board) -> bool:
+        """Handle the entry of a bad subgrid choice"""
+        # TODO need to finish this. maybe move to game logic?
+        # guilty until proven innocent
+        self.good_move = False
+
+        working_subgrid = board.collect_subgrid(str(self.user_select_subgrid))
+        if board.check_grid_victory(working_subgrid) not in ("X", "O"):
+            self.good_move = True
+

```

Comment:
don't forget to check if the board has any empty spaces in it

Reply:
I did forget and then spent over an hour wondering why my logic wasn't working and it's because I was looking for the wrong dot...